projects/storefrontlib/src/cms-components/navigation/search-box/search-box.component.ts

```diff
@@ -131,6 +145,55 @@ export class SearchBoxComponent {
     }
   }
 
+  // Return result list as HTMLElement array
+  private getResultElements(): HTMLElement[] {
+    const suggestions: HTMLElement[] = Array.from(
+      this.winRef.document.querySelectorAll('div.suggestions > a')
+    );
+    const products: HTMLElement[] = Array.from(
+      this.winRef.document.querySelectorAll('div.products > a')
+    );
+
+    return [...suggestions, ...products];
+  }
+
+  // Return focused element as HTMLElement
+  private getFocusedElement(): HTMLElement {
+    return <HTMLElement>this.winRef.document.activeElement;
+  }
+
+  // Focus on previous item in results list
+  focusPreviousChild(event) {
+    event.preventDefault(); // Negate normal keyscroll
```

Comment:
```
event.preventDefault(); // Negate normal keyscroll
const results = this.getResultElements();
const focusedIndex = results.indexOf(this.getFocusedElement());
```that part is duplicated so should be separate to function

Reply:
After discuss with Kacper, every approach causes violates the rules of cc, so let leave as is